src/Microsoft.Graph.Core/Requests/HttpProvider.cs

```diff
@@ -40,14 +39,15 @@ public HttpProvider(ISerializer serializer = null)
         /// </summary>
         /// <param name="httpClientHandler">An HTTP client handler to pass to the <see cref="HttpClient"/> for sending requests.</param>
         /// <param name="disposeHandler">Whether or not to dispose the client handler on Dispose().</param>
+        /// <param name="authenticationProvider">The <see cref="IAuthenticationProvider"/> for authenticating request messages.</param>
         /// <param name="serializer">A serializer for serializing and deserializing JSON objects.</param>
         /// <remarks>
         ///     By default, HttpProvider disables automatic redirects and handles redirects to preserve authentication headers. If providing
         ///     an <see cref="HttpClientHandler"/> to the constructor and enabling automatic redirects this could cause issues with authentication
         ///     over the redirect.
         /// </remarks>
-        public HttpProvider(HttpClientHandler httpClientHandler, bool disposeHandler, ISerializer serializer = null)
```

Comment:
I think we only need to add the authProvider parameter to the ctor that our GraphServiceClient uses.  The long term goal is to phase out HttpProvider so, I don't think we need to change all the overloads.